### PR TITLE
Fix #3075

### LINKED ISSUE #3075: ProcessStartInfoTests.TestEnvironmentOfChildProcess failed on Windows in CI
http://dotnet-ci.cloudapp.net/job/dotnet_corefx_windows_debug_prtest/3613/console
```
Discovering: System.Diagnostics.Process.Tests
  Discovered:  System.Diagnostics.Process.Tests
  Starting:    System.Diagnostics.Process.Tests
     System.Diagnostics.ProcessTests.ProcessStartInfoTests.TestEnvironmentOfChildProcess [FAIL]
        Assert+WrapperXunitException : File path: d:\j\workspace\dotnet_corefx_windows_debug_prtest\src\System.Diagnostics.Process\tests\ProcessStartInfoTests.cs. Line: 220
        ---- Assert.True() Failure
        Expected: True
        Actual:   False
        Stack Trace:
           d:\j\workspace\dotnet_corefx_windows_debug_prtest\src\Common\tests\System\Diagnostics\AssertWithCallerAttributes.cs(582,0): at Assert.WrapException(Exception e, String callerFilePath, Int32 callerLineNumber)
           d:\j\workspace\dotnet_corefx_windows_debug_prtest\src\Common\tests\System\Diagnostics\AssertWithCallerAttributes.cs(577,0): at Assert.True(Boolean condition, String userMessage, String path, Int32 line)
           d:\j\workspace\dotnet_corefx_windows_debug_prtest\src\System.Diagnostics.Process\tests\ProcessStartInfoTests.cs(220,0): at System.Diagnostics.ProcessTests.ProcessStartInfoTests.TestEnvironmentOfChildProcess()
           ----- Inner Stack Trace -----
           d:\j\workspace\dotnet_corefx_windows_debug_prtest\src\Common\tests\System\Diagnostics\AssertWithCallerAttributes.cs(570,0): at Assert.True(Boolean condition, String userMessage, String path, Int32 line)
  Finished:    System.Diagnostics.Process.Tests
```

cc: @Maxwe11, @Priya91, @pallavit

### PR DESCRIPTION
**Pull request by Maxwe11**
fixes #3075

### CODE REVIEW

File: src/Scenarios/tests/InterProcessCommunication/EnvironmetOfChildProcess.cs [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.IO.Pipes;
+using System.Linq;
+using Xunit;
+
+namespace InterProcessCommunication.Tests
+{
+    public class EnvironmetOfChildProcess : RemoteExecutorTestBase
```
Comment (akoeplinger) [inline src/Scenarios/tests/InterProcessCommunication/EnvironmetOfChildProcess.cs]: typo: `Environment` (also in the filename)

**Comment by stephentoub** [comment]
@Maxwe11, what was wrong with the previous test, and how does this one fix it?  Trying to understand why we were seeing an intermittent failure.

**Comment by Maxwe11** [comment]
@stephentoub the problem was with that default `Console` encoding doesn't handle non ASCII chars (e.g. K**ö**plinger). I see the same behavior on Desktop.

**Comment by akoeplinger** [comment]
@Maxwe11 hmm, but why didn't https://github.com/dotnet/corefx/pull/3086 fail with the same error then?

**Comment by stephentoub** [comment]
I'm not convinced the encoding issue is the only issue (and if it was, I'd prefer to see a fix that just set up the ProcessStartInfo and Console encodings correctly rather than rewriting the test entirely).  I just tried adding ```Environment.SetEnvironmentVariable(TempVariableName, "\x1234\x5678");``` to the start of the test... with that, sometimes the test fails, sometimes it passes.  I'd like to better understand why there's an inconsistency before any attempted fix is made.

**Comment by stephentoub** [comment]
I think I know the reason... I'll experiment and post back here.

**Comment by Maxwe11** [comment]
replaced by #3102

**Comment by stephentoub** [comment]
Thanks, @Maxwe11.

**Pull request closed without merging**